buzsaki_lab_to_nwb/valero/stimulilaserinterface.py

```diff
@@ -0,0 +1,152 @@
+from pathlib import Path
+
+import numpy as np
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils.json_schema import FolderPathType
+from pymatreader import read_mat
+from pynwb.epoch import TimeIntervals
+from pynwb.file import NWBFile
+from pynwb.ogen import OptogeneticSeries, OptogeneticStimulusSite
+
+from buzsaki_lab_to_nwb.valero.ecephys_interface import (
+    generate_neurolight_device_metadata,
+)
+
+
+class VeleroOptogeneticStimuliInterface(BaseDataInterface):
+    def __init__(self, folder_path: FolderPathType):
+        super().__init__(folder_path=folder_path)
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False):
+        self.session_path = Path(self.source_data["folder_path"])
+        self.session_id = self.session_path.stem
+
+        pulses_data_path = self.session_path / f"{self.session_id}.pulses.events.mat"
+        assert pulses_data_path.is_file(), f"{pulses_data_path} not found"
+
+        mat_file = read_mat(pulses_data_path)
+        pulses_data = mat_file["pulses"]
+        pulse_intervals = pulses_data["timestamps"]
+        pulse_micro_led = pulses_data["analogChannel"]
+        pulse_amplitude = pulses_data["amplitude"]
+
+        # Create device
+        device_metadata = generate_neurolight_device_metadata()
+        if device_metadata["name"] not in nwbfile.devices:
+            neurolight_probe = nwbfile.create_device(**device_metadata)
+        else:
+            neurolight_probe = nwbfile.devices[device_metadata["name"]]
+        # Create the sites
+        site_description = "microscopic LED 10 x 15 µm each, 3 per shank. Each μLED has an emission area of 150 μm2"
+        location = "dorsal right hippocampus (antero-posterior 2.0 mm, mediolateral 1.5 mm, dorsoventral 0.6 mm)"
+
+        micro_led_ids_to_site = dict()
+        micro_led_ids = np.unique(pulse_micro_led)
+
+        for id in micro_led_ids:
+            optogenetic_site = OptogeneticStimulusSite(
+                name=f"Microled site in Neurolight probe with id {id}",
```

Comment:
I'm against the idea of including spaces in names as they ultimately result as keys in dictionaries (or worse, as attributes) when exploring an NWB file, and it's quite unusual as a practice to use spaces for dictionary keys

@h-mayorquin what do you think? Can we give these more typical object names, like `OptogeneticStimulusSite{id}`?

Reply:
I think this makes sense. Yes. Changing it!